aten/src/ATen/native/AdaptiveAveragePooling.cpp

```diff
@@ -0,0 +1,330 @@
+#include "ATen/ATen.h"
+#include "ATen/NativeFunctions.h"
+#include <tuple>
+
+#define START_IND(a,b,c) (int)std::floor((float)(a * c) / b)
+#define END_IND(a,b,c) (int)std::ceil((float)((a + 1) * c) / b)
```

Comment:
Doesn't have to be this diff, but it would be nice to de-macroify this. I don't think there's any reason we can't do this as an inline function.